tests/EmailValidator/Validation/DNSCheckValidationTest.php

```diff
@@ -14,50 +16,94 @@ public function validEmailsProvider()
-            ['ABC@EXAMPLE.COM'],
-            ['Abc.123@example.com'],
-            ['user+mailbox/department=shipping@example.com'],
-            ['!#$%&\'*+-/=?^_`.{|}~@example.com'],
+            ['Abc@github.com'],
+            ['ABC@GITHUB.COM'],
+            ['Abc.123@github.com'],
+            ['user+mailbox/department=shipping@github.com'],
+            ['!#$%&\'*+-/=?^_`.{|}~@github.com'],
 
             // quoted string
-            ['"Abc@def"@example.com'],
-            ['"Fred\ Bloggs"@example.com'],
-            ['"Joe.\\Blow"@example.com'],
+            ['"Abc@def"@github.com'],
+            ['"Fred\ Bloggs"@github.com'],
+            ['"Joe.\\Blow"@github.com'],
 
             // unicide
             ['ñandu.cl'],
         ];
     }
 
+    public function localOrReservedEmailsProvider()
+    {
+        return [
+            // Reserved Top Level DNS Names
+            ['test'],
+            ['example'],
+            ['invalid'],
+            ['localhost'],
+
+            // mDNS
+            ['local'],
+
+            // Private DNS Namespaces
+            ['intranet'],
+            ['internal'],
+            ['private'],
+            ['corp'],
+            ['home'],
+            ['lan'],
+        ];
+    }
+
     /**
      * @dataProvider validEmailsProvider
      */
-    public function testValidDNS($validEmail)
+    public function testValidDns($validEmail)
     {
         $validation = new DNSCheckValidation();
         $this->assertTrue($validation->isValid($validEmail, new EmailLexer()));
     }
 
-    public function testInvalidDNS()
+    public function testInvalidDns()
     {
         $validation = new DNSCheckValidation();
         $this->assertFalse($validation->isValid("example@invalid.example.com", new EmailLexer()));
     }
 
-    public function testDNSWarnings()
+    /**
+     * @dataProvider localOrReservedEmailsProvider
+     */
+    public function testLocalOrReservedDomainError($localOrReservedEmails)
+    {
+        $validation = new DNSCheckValidation();
+        $expectedError = new LocalOrReservedDomain();
+        $validation->isValid($localOrReservedEmails, new EmailLexer());
+        $this->assertEquals($expectedError, $validation->getError());
+    }
+
+    public function testDomainAcceptsNoMailError()
+    {
+        $validation = new DNSCheckValidation();
+        $expectedError = new DomainAcceptsNoMail();
+        $isValidResult = $validation->isValid("example@example.com", new EmailLexer());
+        $this->assertEquals($expectedError, $validation->getError());
+        $this->assertFalse($isValidResult);
+    }
+
+    /*
+    public function testDnsWarnings()
     {
         $validation = new DNSCheckValidation();
         $expectedWarnings = [NoDNSMXRecord::CODE => new NoDNSMXRecord()];
         $validation->isValid("example@invalid.example.com", new EmailLexer());
         $this->assertEquals($expectedWarnings, $validation->getWarnings());
     }
+    */
 
-    public function testNoDNSError()
```

Comment:
Might be worth flowing this update through all of the tests (so they are all full camel case). It improves the display when you run the tests (e.g. you get `No dns error` instead of `No d n s error` when running PHPUnit)

Reply:
Give it a try, let see how it looks, in a separate PR please.

Reply:
Will do once we have finished this :-)

Reply:
Supposedly is run by Scrutinizr, so run on every PR. Also Psalm. So the code should be widle analysed. Anyway, go for it. All improvements are welcome.